pkg/cluster/cluster.go

```diff
@@ -102,9 +77,8 @@ type Cluster struct {
 	currentProcess   Process
 	processMu        sync.RWMutex // protects the current operation for reporting, no need to hold the master mutex
 	specMu           sync.RWMutex // protects the spec for reporting, no need to hold the master mutex
-
+	ConnectionPooler map[PostgresRole]*connection_pooler.ConnectionPoolerObjects
```

Comment:
So this whole PR is not solely a refactoring but rather a code change that enables separate connection poolers for the master and replicas ?